crates/runtime/src/traphandlers/macos.rs

```diff
@@ -88,9 +88,27 @@ mod mach_addons {
         pub NDR: NDR_record_t,
         pub exception: exception_type_t,
         pub codeCnt: mach_msg_type_number_t,
-        pub code: [i64; 2],
+
+        // Note the usage of a newtype wrapper here which is intended to mirror
+        // how this seems to need to work with C where this structure should
+        // have an alignment of 4. Using `I64Align4` avoid increasing the
+        // alignment to 8 by using `i64` natively.
+        //
+        // Also note that this is a divergence from the C headers which use
+        // `integer_t` here for this field which is a `c_int`. That isn't
+        // actually reflecting reality apparently though because if `c_int` is
+        // used here then the structure is too small to receive a message.
+        //
+        // In the original Gecko sources this was copied from the structure here
+        // is `pragma(pack(4))` which should be the same as an unaligned i64
+        // here which doesn't increase the overall alignment of the structure.
```

Comment:
`repr(packed)` reduces alignment to 1 I think, but it turns out that Rust supports `repr(packed(4))` which exactly matches what C does so I've switched to that instead of the `packed` newtype wrapper.